### ISSUE TITLE
Is final fit overfitting training data? Questions and more questions.

### ISSUE THREAD

**Issue by Pythonatheart** [keras-team/autokeras#1067]
I am using AutoModel for image classification in AutoKeras version 1.0.2:

auto_model.fit(x_train, y_train)

I have noticed that AutoKeras does a final fit using best model found during trials but I can't seem to find information regarding the nature of this final fit. I presume it is fitting the entire training data (as opposed to 80 train:20 val default split of training data for trials). 

However, this often results in outstanding fit (val_loss-0, val_accuracy=1) for the training data but often not fantastic results for test data e.g. in my case (test_loss=0.5115, test_accuracy=0.804).
For example, on epoch 47 of final fit of best model after 10 trials:
Epoch 47/1000
43/43 [==============================] - 46s 1s/step - loss: 4.2737e-04 - accuracy: 1.0000 - **val_loss: 3.8162e-04 - val_accuracy: 1.0000**   

My questions are:
1) Is there an option to exclude final fit? Like the final_fit method on earlier AutoKeras versions?
2) How do I find out the val_loss and accuracy of best model before final fit?
3) Is there a way to visualize training for best model or all models during/after trials?
4) I have noticed some people implements a time_limit during model.fit(), how would this affect the number of trials (default=100)? For example, a 1 hour time limit? - would AK terminate when 1 hour limit is up, even if it has only reached 10 trials (as an example)? 

Thank you to the AutoKeras team for your time and amazing efforts!
